src/TLuaInterpreter.cpp

```diff
@@ -17161,46 +17161,140 @@ void TLuaInterpreter::initIndenterGlobals()
-        luaL_dostring(pIndenterState.get(), QStringLiteral("package.cpath = package.cpath .. ';%1/?.so'")
-                      .arg(QCoreApplication::applicationDirPath())
-                      .toUtf8().constData());
-        luaL_dostring(pIndenterState.get(), QStringLiteral("package.path = package.path .. ';%1/?.lua'")
-                      .arg(QCoreApplication::applicationDirPath())
-                      .toUtf8().constData());
+    // macOS app bundle would like the search path for the binary modules to
+    // also be set to the current binary directory:
+    luaL_dostring(pIndenterState.get(), QStringLiteral("package.cpath = '%1/?.so;' .. package.cpath")
+                  .arg(QCoreApplication::applicationDirPath()).toUtf8().constData());
+#endif
 
-#elif defined(Q_OS_UNIX)
-    // Need to tweak the lua path for the installed *nix case and AppImage builds as well as
-    // to allow running from a shadow build directory (both qmake and cmake).
-    luaL_dostring(pIndenterState.get(), QStringLiteral("package.path = '" LUA_DEFAULT_PATH "/?.lua;%1/?.lua;%1/../3rdparty/?.lua;%1/../../3rdparty/?.lua;%1/../../mudlet/3rdparty/?.lua;' .. package.path")
-                  .arg(QCoreApplication::applicationDirPath())
-                  .toUtf8().constData());
+#if defined(Q_OS_UNIX)
+    // Now also included for the MacOs case so that MacOS developers can build
+    // from within the Qt Creator IDE
+
+    /* Additional paths for the lua package search path - ? is the "package"
+     * name though in the case for the lua code formatter it itself includes
+     * leading path elements of "lcf" and "workshop" in the
+     * require("lcf.workshop.base") case - the lua script being sought is the
+     * "base.lua" file but our paths must give the loction of the "lcf"
+     * directory two parent directories up:
+     * * installed *nix case:
+     *     "LUA_DEFAULT_PATH/?.lua" (if defined and not empty)
+     * * AppImage (subdirectory of executable):
+     *     "<applicationDirectory>/?.lua"
+     * * QMake shadow builds without CONFIG containing "debug_and_release" AND
+     *   "debug_and_release_target" (though the latter one IS there by default):
+     *     "<applicationDirectory>/../3rdparty/?.lua"
+     * * QMake shadow builds with CONFIG containing "debug_and_release" AND
+     *   "debug_and_release_target" (though the latter one IS there by default):
+     *     "<applicationDirectory>/../../3rdparty/?.lua"
+     * * CMake shadow builds
+     *     "<applicationDirectory>/../../mudlet/3rdparty/?.lua"
+     * NOTE: In the cmake case the shadow build directory is alongside the base
+     * directory of the extracted source package so it MUST be extracted into
+     * a known directory name which is assumed here to be "mudlet"!
+     */
+#if defined(LUA_DEFAULT_PATH)
+    if (!QStringLiteral(LUA_DEFAULT_PATH).isEmpty()) {
+        luaL_dostring(pIndenterState.get(), QStringLiteral("package.path = [["
```

Comment:
This seems to be duplicated except it prepends LUA_DEFAULT_PATH if it exists - can be written better.

Reply:
> This seems to be duplicated except it prepends LUA_DEFAULT_PATH if it exists ...

Sort of, it prepends `LUA_DEFAULT_PATH` if it *exists* **and is not an empty string**.

> ... can be written better.

How so?

Reply:
Yes just append the string if it's there.

I've looked the code and the function is now absolutely massive with a tangle of #ifdefs that make the logic really hard to follow and huge comments you have to read. That is not maintainable, we can't have this - we need to have code that everyone finds nice & pleasant to work with.

Reply:
>Yes just append the string if it's there.

That won't work, if the variable is defined but is *an empty string* then it will cause the path to have `/?.lua` included which means look for it in the root directory - on most OSes that location is readable but not writeable for *normal* users. If you refer back to #1616 you will note part of the problems there was that `LUA_DEFAULT_PATH` was getting defined but not set to anything - and that is very hard to detect in the C/C++ preprocessor and cannot be done cleanly.

Reply:
This check can be done at runtime as far as I can see just fine and we'll have much simpler code, no?

Reply:
Well it eliminated some preprocessor conditional lines and still *seems* to work so: I guess so.